rules/expand_template.bzl

```diff
@@ -16,10 +16,21 @@
 """
 
 def _expand_template_impl(ctx):
+    expanded_substitutions = {}
+    for key, value in ctx.attr.substitutions.items():
+        expanded_substitutions[key] = ctx.expand_make_variables(
```

Comment:
ctx.expand_make_variable is marked as "Deprecated" in Bazel documentation. People before me marked it that way and I think we shouldn't introduce new uses.

Please remove it's use.

I believe it's implementable in Starlark, and skylib would be a nice location for such an implementation. (And we might already have one implementation in the C++ rules in builtins).

The rest of the PR seems fine.

Reply:
relevant convo https://groups.google.com/g/bazel-discuss/c/WEnG-WocaTc

so you're saying we should make a new function to do the same thing in skylib and use that here?

Reply:
Yes. I bumped https://github.com/bazelbuild/bazel/issues/5859 to P3.

And I noticed also https://github.com/bazelbuild/bazel-skylib/pull/486 which might already be doing that. Would you like to help in review?

There might be other PRs, I haven't gone through all of them yet.